### ISSUE TITLE
 Rectify docs for MultiLabelSoftMarginLoss

### ISSUE THREAD

**Issue by andy-yangz** [pytorch/pytorch#36217]
## 📚 Documentation

Concern about MultiLabelSoftMarginLoss, I think there is still a problem with its doc.
As @DNGros pointed out, there is a lot of confusion when using MultiLabelMarginLoss and MultiLabelSoftMarginLoss, and he improved the docs. #15863
However, for the docs of MultiLabelSoftMarginLoss it's still not correct.
Since if you set the label as the docs
> Target: (N, C)(N,C) , label targets padded by -1 ensuring same shape as the input.

pad it like [0,3,-1,-1], the result will be wrong.
If we look at the equation
![image](https://user-images.githubusercontent.com/23011317/78770175-cf7d5280-79c0-11ea-85ec-5e7122cd5480.png)
we will find it, in fact, write clearly that label y need to be 0 or 1.
If we feed the -1 padded label, the loss actually sometimes will become negative number, and this is shown by my experiment.

So I think the right Target format for MultiLabelSoftMarginLoss is just one_hot vector.
And we can give a good example code as MultiLabelMarginLoss has.

**Comment by andy-yangz** [comment]
Do I need to submit a pull request to fix this issue?
I have no such open-source contributing experience before, so I wish to get some guidance.

**Comment by shaibagon** [comment]
Just came across [this](https://stackoverflow.com/q/66979824/1714410) SO question raising the same issue.

**Comment by zhiqihuang** [comment]
I think the target of this loss function should be binary vectors indicating the positive class. The doc of this loss function is somehow the same as the [MultiLabelMarginLoss](https://pytorch.org/docs/stable/generated/torch.nn.MultiLabelMarginLoss.html#torch.nn.MultiLabelMarginLoss).  But MultiLabelSoftMarginLoss is more like the multi-label version of cross-entropy.
